### PR TITLE
.travis.yml for running node tests

### PR DESCRIPTION
**Pull request by elimisteve**
@ajvb How can this be made better?

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/cryptag/minishare/miniware"
+)
+
+func TestLogin(t *testing.T) {
+	r := NewRouter(miniware.NewMapper())
+
+	testURL(t, "GET", "/api/login", nil, r,
+		http.StatusBadRequest, `{"error":"Error: invalid miniLock ID"}`)
+
+	for _, url := range []string{"/dashboard", "/pursuance", "/123"} {
```
Comment (ajvb) [inline server_test.go]: Want to move these out of `TestLogin` to their own specific testing functions?

File: server_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	if headers != nil {
+		req.Header = headers
+	}
+
+	rec := httptest.NewRecorder()
+
+	handler.ServeHTTP(rec, req)
+
+	if wantedStatusCode != 0 {
```
Comment (ajvb) [inline server_test.go:45]: This is totally fine, but I'm a big fan of the `assert` library in https://github.com/stretchr/testify

**Comment by ajvb** [comment]
This is a great start. The next piece imo would be getting either CodeClimate or something else to provide test coverage metrics. For kala I use Coverall's - https://coveralls.io/github/ajvb/kala?branch=master - but might be worth checking out CodeClimate's test coverage thing if we are going to use them for other analysis.

File: server_test.go (hunk 1 above)
Comment (ajvb) [reply server_test.go]: Same with the `404` check below

**Comment by elimisteve** [comment]
Example of how to use Travis with multiple languages: https://stackoverflow.com/a/44054333/197160

**Comment by elimisteve** [comment]
Another example: https://github.com/travis-ci/travis-ci/issues/4090#issuecomment-184811689

**Pull request merged**